Assets/Scripts/Puzzle/Lever.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Lever : SignalEmitter
+{
+	private Animator animator;
+
+	public bool isActive = false;
```

Comment:
The signalEmitter superclass already have a isActivated bool and you are using its property. You are correct this can probably be removed.

Reply:
not sure if this could be removed, it might seem that it's not used, but it's needed for the PlayerInteract script to see and use this to see how to use the lever when pressing use-button..